src/components/NavBar.jsx

```diff
@@ -1,17 +1,25 @@
 import React from 'react'
+import { useSelector } from 'react-redux'
 import { NavLink } from 'react-router-dom'
+import { isLoggedIn, getRole } from '../modules/auth-module'
+import { renderSignUpButton, renderAddHotel, renderDeleteHotel, renderMyReservations, renderReserve } from '../modules/conditionalRendering'
+
+function NavBar() {
+
+const token = useSelector(state => state.login.token)
+const loggedIn = isLoggedIn(token)
+const role = getRole(token)
 
-function NavBar({loggedIn} = loggedIn) {
   return (
     <nav className='w-[20vw] h-[100vh] flex flex-col items-center justify-evenly py-4 border-r'>
       <a href='/'><img src="Hotelzilla-logo.png" alt="Hotelzilla Logo" className='h-[25vh]'/></a>
       <ul className='flex flex-col mt-12'>
-         <NavLink to="/hotels/add" className="py-2 font-bold hover:bg-lime-400">ADD HOTEL</NavLink>
-         <NavLink to="/hotels/delete" className="py-2 font-bold hover:bg-lime-400">DELETE HOTEL</NavLink>
-         <NavLink to="/hotels/reserve" className="py-2 font-bold hover:bg-lime-400">RESERVE</NavLink>
-         <NavLink to="/reservations" className="py-2 font-bold hover:bg-lime-400">MY RESERVATIONS</NavLink>
-         <NavLink to="/register" className="py-2 font-bold hover:bg-lime-400">SIGN UP</NavLink>
-         <NavLink to="/login" className="py-2 font-bold hover:bg-lime-400">LOG IN</NavLink>
+         { renderAddHotel(role) }
+         { renderDeleteHotel(role) }
+         { renderReserve(loggedIn) }
+         { renderMyReservations(loggedIn) }
+         <NavLink to="/login" className="py-2 font-bold hover:bg-lime-400 text-xl font-[Taxicab]">{loggedIn ? "LOG OUT" : "LOG IN"}</NavLink>
+         { renderSignUpButton(loggedIn) }
```

Comment:
- [ ] [Optional] Nice approach to solve the roles and to check if a user is logged or not, you can also use a conditional rendering directly and pair together the properties that share that condition, it could be like:
``` javascript
...
{ role='admin' && (
  <Navlink>...</Navlink>
  <Navlink>...</Navlink>
)}
{ user='logged' && (
  <Navlink>...</Navlink>
  <Navlink>...</Navlink>
  <Navlink>...</Navlink>
)}
...
```

Reply:
@williamrolando88, excellent, I'll change it 👍